internals/secrethub/service_init.go

```diff
@@ -151,3 +123,40 @@ func (cmd *ServiceInitCommand) Register(r Registerer) {
 
 	BindAction(clause, cmd.Run)
 }
+
+func givePermission(service *api.Service, repo api.RepoPath, permissionFlagValue string, client secrethub.Client) error {
```

Comment:
Already a lot better. But I still kind of miss an explanation of what's happening with the `SpltN` (that the path part is optional). Fortunately this can now be easily added in the comment of this function.